src/goto-programs/assign_params_as_non_det.cpp

```diff
@@ -36,24 +57,217 @@ bool assign_params_as_non_det::runOnFunction(
       return false; // Not expected
     exprt lhs = symbol_expr(*context.find_symbol(_id));
 
-    // rhs
-    exprt rhs = exprt("sideeffect", lhs.type());
-    rhs.statement("nondet");
-    rhs.location() = l;
+    typet l_t = lhs.type();
+    if (l_t.is_pointer() && l_t.subtype() != empty_typet())
+    {
+      /*
+        e.g. int* lhs;
+        to
+        lhs = null;
+        bool temp;
+        temp = nondet();
+        if(temp)
+        {
+          int temp2;
+          temp2 = nondet();
+          assume(temp2 != 0);
+          lhs = &temp2;
+        }
+        
```

Comment:
Would you mind making your instrumentation comments match the name of these variables?

Reply:
I change the comments instead